pkg/api/dashboard_public.go

```diff
@@ -49,8 +50,39 @@ func (hs *HTTPServer) SavePublicDashboardConfig(c *models.ReqContext) response.R
 	return response.JSON(http.StatusOK, pdc)
 }
 
+// QueryPublicDashboard returns all results for the panels on a public dashboard
+// POST /api/public/dashboard/:publicUid/query
+func (hs *HTTPServer) QueryPublicDashboard(c *models.ReqContext) response.Response {
+	timeRangeDTO := dtos.TimeRangeOnlyMetricRequest{}
+	if err := web.Bind(c.Req, &timeRangeDTO); err != nil {
+		return response.Error(http.StatusBadRequest, "bad request data", err)
+	}
+
+	dashboard, _ := hs.dashboardService.GetPublicDashboard(c.Req.Context(), web.Params(c.Req)[":uid"])
+
+	queries := models.GetQueriesFromDashboard(dashboard.Data)
+
+	reqDTO := dtos.MetricRequest{
+		To:      timeRangeDTO.To,
+		From:    timeRangeDTO.From,
+		Queries: nil,
+	}
+
+	for _, panel := range queries {
+		for _, query := range panel {
+			reqDTO.Queries = append(reqDTO.Queries, query)
+		}
+	}
+
+	resp, err := hs.queryDataService.QueryData(c.Req.Context(), c.SignedInUser, c.SkipCache, reqDTO, true)
```

Comment:
c.SignedInUser will always be anonymous. Maybe we should explicitly define it's value. Nil?